index.js

```diff
@@ -151,6 +151,10 @@ module.exports = function (newrelic, opts) {
 	}
 
 	return async function koaNewrelic(ctx, next) {
+		if (opts.preSetTransaction) {
+			setTransactionName(ctx.method, ctx.path);
+		}
+
 		await next();
 
 		if (ctx._matchedRoute) {
```

Comment:
then have u checked why your `koa-router` cannot set `_matchedRoute` for you to group?